src/main/java/org/wildfly/swarm/plugin/repository/RepositoryBuilderMojo.java

```diff
@@ -142,6 +150,33 @@ private void executeGeneratedProjectBuild(File pomFile, File projectDir, File re
         getLog().info("Built project from BOM: " + projectDir.getAbsolutePath());
     }
 
+    private void addBom() throws MojoFailureException, IOException {
+        getLog().info("Add the bom to the repository");
+
+        String[] groupIdParts = project.getGroupId().split("\\.");
+        File groupIdDir = createSubDirectory(repoDir, groupIdParts);
+        File targetDir = createSubDirectory(
+                groupIdDir, project.getArtifactId(), project.getVersion()
+        );
+        String fileName = String.format("%s-%s.%s",
+                project.getArtifactId(), project.getVersion(), "pom"
+        );
+
+        File targetFile = new File(targetDir, fileName);
+
+        FileUtils.copyFile(getBomFile(), targetFile);
+        getLog().info("Copied the bom to " + targetFile.getAbsolutePath());
+    }
+
+    private File createSubDirectory(File repoDir, String... subPath) {
```

Comment:
Hm, this method actually does not create any directory. It only creates a `File` instance and could be probably replaced with `java.nio.file.FileSystem.getPath(String, String...).toFile()`?

Reply:
yes, copyFile creates the required directories. I'll try replacing `createSubDirectory` with `getPath`.

Reply:
Done